cpu/sam0_common/periph/i2c.c

```diff
@@ -104,6 +191,11 @@ void i2c_init(i2c_t dev)
     const uint32_t fSCL = i2c_config[dev].speed;
     const uint32_t fGCLK = sam0_gclk_freq(i2c_config[dev].gclk_src);
 
+    /* initial check if bus is blocked & resolve attempt */
+    if (!_check_and_unblock_bus(&i2c_config[dev], false)) {
```

Comment:
Why not 

```suggestion
    if (!_check_and_unblock_bus(dev, false)) {
```

then you can drop the `config()` helper - you'll also have to do that for `_i2c_start()`, but it'll be a trivial change.

Reply:
I think I addressed this in the way you suggested.